pkg/envoy/rds/route/route_config.go

```diff
@@ -286,6 +287,40 @@ func buildWeightedCluster(weightedClusters mapset.Set) *xds_route.WeightedCluste
 	sort.Stable(clusterWeightByName(wc.Clusters))
 	return &wc
 }
+func buildRetryPolicy(retry *v1alpha1.RetryPolicySpec) *xds_route.RetryPolicy {
+	rp := &xds_route.RetryPolicy{}
+	if retry == nil {
+		return rp
```

Comment:
Return nil object instead of pointer to an empty object.
```suggestion
		return nil
```